lib/cisco_node_utils/interface.rb

```diff
@@ -578,13 +578,20 @@ def vrf
       config_get('interface', 'vrf', name: @name)
     end
 
-    def vrf=(vrf)
-      fail TypeError unless vrf.is_a?(String)
-      if vrf.empty?
+    def vrf=(v)
+      fail TypeError unless v.is_a?(String)
+      return if v == vrf
```

Comment:
Call to the `vrf` instance method to get the current VRF setting - so that we can correctly do a no-op.